interchain.go

```diff
@@ -442,9 +444,11 @@ type relayerChain struct {
 // BuildWallet will generate a random key for the key name in the provided keyring.
 // Returns the mnemonic and address in the bech32 format of the provided ChainConfig.
 func BuildWallet(kr keyring.Keyring, keyName string, config ibc.ChainConfig) ibc.Wallet {
-	// NOTE: this is hardcoded to the cosmos coin type.
-	// In the future, we may need to get the coin type from the chain config.
-	const coinType = types.CoinType
+	coinTypeU64, err := strconv.ParseUint(config.CoinType, 10, 32)
```

Comment:
The variable is named `coinTypeU64` but you are telling `ParseUInt` to use 32bits, and then on L451 there's a typecast to `uint32`. It looks like the coin type is a `uint32` so I'd say this parse call is fine but the type cast could be eliminated. 

Also the name could be shortened to just `coinType` for brevity

Reply:
`ParseUInt` outputs a `uint64` whether you specify 32bits or 64bits.
Regardless, for readability, I like what you're saying here and can do the type cast in the `hd.CreateHdPath` function (L456) with the renaming suggestion.

Reply:
Well TIL! Disregard my comment then 😛